src/app/effects/knowledge.ts

```diff
@@ -45,15 +45,27 @@ export class KnowledgeEffects {
             const res = response.query.pages;
             const pageID = Object.keys(res)[0];
        if (res[pageID].extract) {
-        this.store.dispatch(new knowledge.SearchAction(res[pageID]));
+        this.store.dispatch(new knowledge.SearchContent(res[pageID]));
         return empty();
        } else {
-                   this.store.dispatch(new knowledge.SearchAction([]));
+                   this.store.dispatch(new knowledge.SearchContent([]));
                     return empty();
             }
 
 
         });
+        this.knowledgeservice.getImage(querypay.query)
+        .takeUntil(nextSearch$)
+        .subscribe((response) => {
+          if (response.RelatedTopics.length > 0) {
+            this.store.dispatch(new knowledge.SearchImage(response));
```

Comment:
What is the type of response here? If it remains fixed, then make a change from ```any``` to the response type in action payload in the action defined above.